coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java

```diff
@@ -279,6 +287,23 @@ public RexNode visitCall(RexCall call) {
       return Optional.empty();
     }
 
+    private Optional<RexNode> visitCast(RexCall call) {
+      final SqlOperator op = call.getOperator();
+      if (op.getKind() != SqlKind.CAST) {
+        return Optional.empty();
+      }
+
+      RexNode leftOperand = call.getOperands().get(0);
```

Comment:
I think we need to use
```
  List<RexNode> convertedOperands = visitList(call.getOperands(), (boolean[]) null);
  RexNode leftOperand = convertedOperands.get(0);
```
here to take care of the possible nested transformations.

For example, for SQL
```
select cast(from_utc_timestamp(2592000.0,'PST') as decimal(10, 0))
```
we also need to transform the inner `from_utc_timestamp`.

Reply:
Good catch, thanks! Fixed this in the revision and added a unit test for this case as well.